sips/sip-011/sip-011-catastrophic-failure-recovery.md

```diff
@@ -0,0 +1,464 @@
+liveness or safety failure.  A subsequent SIP may codify other types of soft
+forks and their applications in different contexts.  Due to the comparatively
+low barrier-to-entry for mining on the Stacks chain, miner-activated soft forks
+are expected to offer an acceptable measurement of the ecosystem's sentiments on
+a pending soft fork.
+
+A miner-activated soft fork procedure for healing a catastrophic failure works
+as follows:
+
+1. The Foundation and at least one other significant ecosystem entity will approve and
+sign a Stacks node software release that implements the soft fork rules.  This
+is the first step in this process because this new node software must address
+whatever catastrophic bug that led to the need to follow this procedure.  The
+remainder of this procedure is focused on codifying the new rules via the SIP
+process in order to require them in future node software releases.  The act of
+releasing this new node software will follow the critical bugfix point-release
+procedure, but with an additional warning from the announcement email that this
+new release is the beginning of a soft-fork roll-out.
+2. Once the catastrophic bug has been addressed, and a reliable majority of
+blocks on the canonical fork are produced by miners running the new software, a
+SIP describing the soft fork will be written and published describing the new
+rules and why they are necessary.  This SIP must then be advanced in the SIP
+process to "activation-in-progress" status by the relevant advisory boards and
+technical steering committee.  Among other things, this SIP's activation
+criteria must require the following:
+
+   * At least 80% of all miners must _explicitly and continuously vote to
+     accept_ the soft fork rules for at least _two consecutive_ whole PoX reawrd
+cycles. Additional, stricter criteria are permitted.
+   * An _activation window_, defined by a _start activation height_ and an _end
+     activation height_, represented as burnchain block heights, must be
+stipulated in the activation criteria.  The aforementioned miner vote must take
+place and run to completion in-between these two block heights.  The block
+heights _must_ correspond to the beginnings of PoX reward cycles, and must be
+between 14 reward cycles (1 maximum Stacking period plus two reward cycles) and
+25 reward cycles (about 1 year) apart.
+   * A soft-fork-specific bit pattern must be specificied, which miners _must_
+     include in their coinbase transactions in order to indicate unambiguous
+support for activating the new rules.
+
+3. Once a release with the soft fork voting rules and activation window is
+written and adequately tested (possibly including rolling it out to one or more
+public testnets first), it will be announced via a signed email from
+announce@stacks.org.  This release announcement must occur before the activation
+window begins.
+4. If at least 80% of all miners vote to accept the soft fork rules for two
```

Comment:
How do miners vote? Is it by simply continuing to mine with the version that introduces the soft fork?

Reply:
It's mentioned just above this line:

```
   * A soft-fork-specific bit pattern must be specificied, which miners _must_
     include in their coinbase transactions in order to indicate unambiguous
     support for activating the new rules.
```

Reply:
I clarified this a bit more in 3832ae2.